TrafficCapture/trafficCaptureDependencies/build.gradle

```diff
@@ -0,0 +1,107 @@
+
+    def resilience4j = '1.7.0'
+    api group: 'io.github.resilience4j', name: 'resilience4j-ratelimiter', version: resilience4j
+    api group: 'io.github.resilience4j', name: 'resilience4j-retry', version: resilience4j
+
+    def jackson = '2.16.2'
+    api group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: jackson
+    api group: 'com.fasterxml.jackson.dataformat', name: 'jackson-dataformat-smile', version: jackson
+
+    def jupiter = '5.9.3'
+    api group: 'org.junit.jupiter', name:'junit-jupiter-api', version: jupiter
+    api group: 'org.junit.jupiter', name:'junit-jupiter-params', version: jupiter
+    api group: 'org.junit.jupiter', name:'junit-jupiter-engine', version: jupiter
+
+    api group: 'io.burt', name: 'jmespath-core', version: '0.6.0'
+
+    api group: 'org.apache.commons', name: 'commons-compress', version: '1.26.0'
+
+    def awssdk = '2.25.19'
+    api group: 'software.amazon.awssdk', name: 'arns', version: awssdk
+    api group: 'software.amazon.awssdk', name: 'auth', version: awssdk
+    api group: 'software.amazon.awssdk', name: 'sdk-core', version: awssdk
+    api group: 'software.amazon.awssdk', name: 'secretsmanager', version: awssdk
+
+    api group: 'com.bazaarvoice.jolt', name: 'jolt-core', version: '0.1.7'
+
+    def opentelemetry = '1.34.1'
+    api group: 'io.opentelemetry', name: 'opentelemetry-api', version: opentelemetry
+    api group: 'io.opentelemetry', name: 'opentelemetry-exporter-otlp', version: opentelemetry
+    api group: 'io.opentelemetry', name: 'opentelemetry-sdk', version: opentelemetry
+    api group: 'io.opentelemetry', name: 'opentelemetry-sdk-testing', version: opentelemetry
+
+    api group: 'io.opentelemetry.semconv', name: 'opentelemetry-semconv', version: '1.23.1-alpha'
+
+    api group: 'com.lmax', name: 'disruptor', version: '4.0.0'
+
+    def bouncycastle = '1.78'
+    api group: 'org.bouncycastle', name: 'bcprov-jdk18on', version: bouncycastle
+    api group: 'org.bouncycastle', name: 'bcpkix-jdk18on', version: '1.78'
+
+    def testcontainers = '1.19.7'
+    api group: 'org.testcontainers', name: 'junit-jupiter', version: testcontainers
+    api group: 'org.testcontainers', name: 'kafka', version: testcontainers
+    api group: 'org.testcontainers', name: 'testcontainers', version: testcontainers
+    api group: 'org.testcontainers', name: 'toxiproxy', version: testcontainers
+
+    def mockito = '4.6.1'
+    api group: 'org.mockito', name:'mockito-core', version: mockito
+    api group: 'org.mockito', name:'mockito-junit-jupiter', version: mockito
+
+    api group: 'eu.rekawek.toxiproxy', name: 'toxiproxy-java', version: '2.1.7'
+
+    // ************************************************************
+    // The following constraints are for mitigating downstream CVEs
+    // ************************************************************
+    implementation group: 'net.minidev', name: 'json-smart', version: '2.4.9'
```

Comment:
what's the difference here (as constraints) between api and implementation?

Reply:
I needed to get more details for this myself, `api` means pull in the dependency with transitive dependencies.  Whereas `implementation` only changes the specific referenced version, no dependencies.

When we have an `api` dependency with an older transitive dependency with a CVE, by adding the `implementation` constraint it forces the dependency to have the newer version.